src/e3/testsuite/driver/adacore.py

```diff
@@ -154,15 +156,19 @@ def get_script_command_line(self) -> List[str]:
             if support_dir and os.path.isfile(support_script):
                 new_script.append(". $TEST_SUPPORT_DIR/support.sh")
 
-            # Read all lines in the original test script. Get rid of potential
-            # whitespaces and CR at the end of each line, and convert the "cmd"
-            # syntax to Bourne shell.
+            # Read all lines in the original test script.
             with open(self.script_file) as f:
-                for line in f:
-                    line = line.rstrip()
-                    for pattern, replacement in self.cmd_substitutions:
-                        line = pattern.sub(replacement, line)
-                    new_script.append(line)
+                if must_convert:
+                    # Get rid of potential whitespaces and CR at the end of
+                    # each line, and convert the "cmd" syntax to Bourne shell.
+                    for line in f:
+                        line = line.rstrip()
+                        for pattern, replacement in self.cmd_substitutions:
+                            line = pattern.sub(replacement, line)
+                        new_script.append(line)
+                else:
+                    for line in f:
+                        new_script.append(line.rstrip())
```

Comment:
A lot of logic is duplicated. Could you move the check on `must_convert` only to cover the `for` loop on substitutions?